### ISSUE TITLE
Not only the Home Assistant frozen, but also RPI

### ISSUE THREAD

**Issue by Bagunda** [qrioniclabs/homeassistant-mysql_command#2]
All are updated

**Comment by qrioniclabs** [comment]
Hi!

First of all, thanks for trying out the component!

From what you wrote, I can only guess that the SQL server makes the components command wait because another process locks the database. You could use "SHOW PROCESSLIST;" on the server to check on the running processes (look for "Waiting for table metadata lock").

It would be useful if you could state the command that you tried to execute. Can you check HA logs if there are any error messages regarding the component?

I am unsure how Home Assistant handles freezing calls from within a custom component. One should think that it wouldnt allow it to freeze HA or even the whole system...

I will investigate further, maybe I need to intruduce a timeout for the MySQL call.

**Comment by qrioniclabs** [comment]
_Issue closed accidentally. Reopened._

**Comment by qrioniclabs** [comment]
@Bagunda the function used to send the MySQL command has an optional parameter for a connection_timeout. Please try the above patch (just add 1 line to notify.py as seen here: https://github.com/qrioniclabs/homeassistant-mysql_command/pull/3/files) and try again. If the command is put on hold by the MySQL server, the call should now terminate after 10 seconds, so HA and the RPI should not freeze.

**Comment by qrioniclabs** [comment]
Fixed in release 1.1.0

**Comment by Bagunda** [comment]
RPI freezes tightly. It was only possible to write "reboot" in the console. But each letter of the word "reboot" appeared once a minute. Total - only after 10 minutes the command worked.
Therefore, going into MySQL and writing "SHOW PROCESSLIST" is a completely impossible task!

**Comment by qrioniclabs** [comment]
Okay, thanks for your feedback.

So, have you tried the fix or the latest version? Which command have you been trying to execute?

I am unsure if you tested it with the timeout-fix but at the moment I believe this solves you issue.
